htlcswitch/hop/error_encryptor.go

```diff
@@ -94,16 +102,21 @@ type SphinxErrorEncrypter struct {
 //     OnionProcessor.
 func NewSphinxErrorEncrypterUninitialized() *SphinxErrorEncrypter {
 	return &SphinxErrorEncrypter{
-		OnionErrorEncrypter: nil,
-		EphemeralKey:        &btcec.PublicKey{},
+		EphemeralKey: &btcec.PublicKey{},
 	}
 }
 
 func NewSphinxErrorEncrypter(ephemeralKey *btcec.PublicKey,
-	sharedSecret sphinx.Hash256) *SphinxErrorEncrypter {
+	sharedSecret sphinx.Hash256,
+	fatErrors *sphinx.FatErrorStructure) *SphinxErrorEncrypter {
 
 	encrypter := &SphinxErrorEncrypter{
 		EphemeralKey: ephemeralKey,
+		fatErrors:    fatErrors,
+	}
+
+	if fatErrors != nil {
+		encrypter.CreatedAt = time.Now()
```

Comment:
can set this field regardless of fat errors active or not

Reply:
or it has to do with the serialization logic that we don't want to set it in that case?

Reply:
Yes the serialization logic, but indirectly. It wouldn't be a problem to set it, because serialization is gated on `s.fatErrors` which is still `nil`. Unit tests and their `DeepEqual` calls will fail on the creation time mismatch. Tried to make the comparison more sophisticated, but didn't turn out that nice. So decided to just not set `CreatedAt` here.